src/main/java/org/broadinstitute/hellbender/engine/ReadsHtsgetDataSource.java

```diff
@@ -0,0 +1,391 @@
+     * {@link #setTraversalBounds}
+     *
+     * @param interval The interval over which to query
+     * @return Iterator over reads overlapping the query interval
+     */
+    @Override
+    public Iterator<GATKRead> query(final SimpleInterval interval) {
+        return this.prepareIteratorsForTraversal(Collections.singletonList(interval), false);
+    }
+
+    /**
+     * @return An iterator over just the unmapped reads with no assigned position. This operation is not affected
+     *         by prior calls to {@link #setTraversalBounds}. The underlying file must be indexed.
+     */
+    @Override
+    public Iterator<GATKRead> queryUnmapped() {
+        return this.prepareIteratorsForTraversal(null, true);
+    }
+
+    private Iterator<GATKRead> prepareIteratorsForTraversal(final List<SimpleInterval> intervals, final boolean traverseUnmapped) {
+        final Map<SamReader, CloseableIterator<SAMRecord>> mapping = intervals == null
+            ? this.getReaders(traverseUnmapped)
+            : this.getReadersWithIntervals(intervals, traverseUnmapped);
+
+        if (mapping.isEmpty()) {
+            return Collections.emptyIterator();
+        }
+        final CloseableIterator<SAMRecord> samRecordIterator;
+        if (mapping.size() == 1) {
+            samRecordIterator = mapping.values().iterator().next();
+        } else {
+            final Set<SAMFileHeader> headers = mapping.keySet().stream().map(SamReader::getFileHeader).collect(Collectors.toSet());
+            samRecordIterator = new MergingSamRecordIterator(this.createHeaderMerger(headers), mapping, true);
+        }
+        this.iterators.add(samRecordIterator);
+        return new SAMRecordToReadIterator(samRecordIterator);
+    }
+
+    private Map<SamReader, CloseableIterator<SAMRecord>> getReaders(final boolean traverseUnmapped) {
+        final Map<SamReader, CloseableIterator<SAMRecord>> mapping = new ConcurrentHashMap<>(this.sources.size());
+        final List<Future<?>> futures = new ArrayList<>(this.sources.size());
+        this.sources.parallelStream()
+            .forEach(source -> futures.add(this.executorService.submit(() -> {
+                final HtsgetRequest req = new HtsgetRequest(source);
+                if (traverseUnmapped) req.setInterval(HtsgetRequest.UNMAPPED_UNPLACED_INTERVAL);
+                final SamReader reader = this.readerFactory.open(SamInputResource.of(req.getResponse().getDataStream()));
+                mapping.put(reader, this.wrapIteratorWithClose(reader.iterator(), reader));
+            })));
+        try {
+            for (final Future<?> future : futures) future.get();
+        } catch (final ExecutionException | InterruptedException e) {
+            throw new UserException("Interrupted while initializing iterator", e);
+        }
+        return mapping;
+    }
+
+    private Map<SamReader, CloseableIterator<SAMRecord>> getReadersWithIntervals(final List<SimpleInterval> intervals, final boolean traverseUnmapped) {
```

Comment:
In this case traverseUnmapped means what I think it does, but I might rename it to "includeUnplacedUnmapped"